zap/zap.go

```diff
@@ -14,89 +15,173 @@ var (
 	_ slog.Logger = (*Logger)(nil)
 )
 
-// LoggerEntry is either a zap.Logger or a zap.SugaredLogger
-type LoggerEntry interface{}
-
 // Logger is an adaptor using go.uber.org/zap as slog.Logger
 type Logger struct {
-	entry LoggerEntry
+	logger *zap.Logger
+	cfg    *zap.Config
 }
 
 // Enabled tells this logger is enabled
-func (zl *Logger) Enabled() bool {
+func (zpl *Logger) Enabled() bool {
+	if zpl == nil || zpl.logger == nil || zapcore.LevelOf(zpl.logger.Core()) == zapcore.InvalidLevel {
+		return false
+	}
+
 	return true
```

Comment:
shouldn't we be calling `zpl.logger.Check()` here to confirm if a log entry of this level would be logged?

Reply:
The "core" has a minimum level, that is what we check here

Reply:
Enabled() isn't only avoid being _valid_. it's intended to check if the level specified via `WithLevel()` will be processed and it's worth the processing effort